### DISCUSSION TITLE
pipewire_output: Decoder is too slow; playing silence to avoid xrun [one solution]

### DISCUSSION THREAD

**Discussion by headmidtail** [MusicPlayerDaemon/MPD#1896] [General]
Since this error message seems to appear quite frequently, I want to express my experience as a possible hint. Though the currently open issue seems to has another cause, hope it helps anyway:

I've got this "Decoder is too slow; playing silence to avoid xrun" for a long time now, with several versions and on different hardware. Earlier with "alsa_output:" and currently with "pipewire_output:". Resulting in short interruptions on some songs, mostly in the first 10 seconds, but not always. Everything on archlinux. I've recently seen that journalctl showed a very few "exception: Failed to open "HTTP Stream" (httpd); error initializing vorbis encoder" as well.
 
Therefore I've exchanged the vorbis encoder with lame in the httpd type audio_output section and both errors are gone, finally.
```
audio_output {
        type            "httpd"
        name            "HTTP Stream"
#       encoder         "vorbis"                # optional, vorbis or lame
        encoder         "lame"                  # optional, vorbis or lame
        port            "8000"
        bind_to_address "0.0.0.0"               # optional, IPv4 or IPv6
        bitrate         "128"                   # do not define if quality is defined

}```

I now there are different reasons for this error message, but just want to share.

**Comment by skidoo23** [comment]
Do you have multiple audio_outputs enabled? Subject is "pipewire..." and your config is about httpd. Please enable with your favorite mpd client only ONE (1) audio_output at the same time.

**Reply by headmidtail** [reply]
Thanks, after turning off httpd output, the small lags seem to be gone (need to test it a little bit longer). Nevertheless it won't help s.o. who wants to stream in parallel.

**Reply by skidoo23** [reply]
For multiple outputs in sync use MPD SnapCast output plugin and SnapCast player for every audio output.

@MaxKellermann maybe related to https://github.com/MusicPlayerDaemon/MPD/issues/356 ?
